pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/Helper.java

```diff
@@ -211,11 +211,16 @@ static String getFQVariableName(final ASTNewKeyValueObjectExpression variable) {
 
     static boolean isSystemLevelClass(ASTUserClass node) {
         List<TypeRef> interfaces = node.getNode().getDefiningType().getCodeUnitDetails().getInterfaceTypeRefs();
+
+        boolean hasWhitelistedInterfaces = false;
         for (TypeRef intObject : interfaces) {
-            return isWhitelisted(intObject.getNames());
+            if (isWhitelisted(intObject.getNames())) {
+                hasWhitelistedInterfaces = true;
```

Comment:
`return true` would be cleaner... or am I missing the point of the rule ?

Reply:
yes, it's not really clear. If I remember correctly, the rule that triggered, was [AvoidBranchingStatementAsLastInLoop](https://pmd.github.io/pmd-6.1.0/pmd_rules_java_errorprone.html#avoidbranchingstatementaslastinloop). And looking now again at the original code - that's actually a bug - it would have only checked the first interface of the list.

Yes, I think, using just `return true` here and avoiding the local variable should work...